### PR TITLE
Add Rook version 1.7.11

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What type of PR is this?

type::chore

#### What this PR does / why we need it:

Adds support for Rook version 1.7.11

#### Which issue(s) this PR fixes:

Fixes NONE

#### Does this PR introduce a user-facing change?

```release-note
- Adds [Rook add-on](https://kurl.sh/docs/add-ons/rook) version 1.7.11.
  - Upgrades Ceph cluster from Octopus to [Pacific](https://docs.ceph.com/en/quincy/releases/pacific/).
- Fixes an issue that causes Rook upgrades to fail on single node installations because of Rook MDS pod anti-affinity rules.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: addons/rook/template/base/install.sh [hunk 1]
```diff
@@ -143,13 +148,6 @@ function rook_cluster_deploy() {
     local src="${DIR}/addons/rook/${ROOK_VERSION}/cluster"
     local dst="${DIR}/kustomize/rook/cluster"
 
-    # Don't redeploy cluster - ekco may have made changes based on num of nodes in cluster
-    if kubectl -n rook-ceph get cephcluster rook-ceph >/dev/null 2>&1 ; then
-        echo "Cluster rook-ceph already deployed"
-        rook_cluster_deploy_upgrade
-        return 0
-    fi
-
```
Comment (laverya) [inline addons/rook/template/base/install.sh:152]: Are you sure this won't be undoing changes made by ekco?
Or are we just fine with those?

File: addons/rook/template/base/install.sh [hunk 2]
```diff
@@ -209,7 +222,7 @@ function rook_cluster_deploy_upgrade() {
         rook_versions="$(kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{"rook-version="}{.metadata.labels.rook-version}{"\n"}{end}' | sort | uniq)"
         if [ -n "${rook_versions}" ] && [ "$(echo "${rook_versions}" | wc -l)" -gt "1" ]; then
             logWarn "Detected multiple Rook versions"
-            logWarn "${rook_versions}"
+            kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}name={.metadata.name}, rook-version={.metadata.labels.rook-version}{"\n"}{end}'
```
Comment (laverya) [inline addons/rook/template/base/install.sh:225]: A much nicer output than before

File: addons/rook/template/testgrid/k8s-docker.yaml [hunk 3]
```diff
@@ -69,19 +77,23 @@
...
       version: "latest"
-    docker:
+    containerd:
       version: "latest"
     minio:
       version: "latest"
     kotsadm:
       version: "latest"
+    ekco:
+      version: "latest"
```
Comment (laverya) [inline addons/rook/template/testgrid/k8s-docker.yaml:96]: Do we still have any tests without ekco?

**Review by laverya — DISMISSED** [review]

File: addons/rook/template/testgrid/k8s-docker.yaml (hunk 3 above)
Comment (emosbaugh) [reply addons/rook/template/testgrid/k8s-docker.yaml:96]: No but I'd like to say at this point ekco is required and eventually make it core

**Review by laverya — APPROVED** [review]

**Pull request merged**